src/pkg/robot/controller.go

```diff
@@ -58,17 +58,22 @@ func (d *DefaultAPIController) GetRobotAccount(id int64) (*model.Robot, error) {
 func (d *DefaultAPIController) CreateRobotAccount(robotReq *model.RobotCreate) (*model.Robot, error) {
 
 	var deferDel error
-	// Token duration in minutes
+
 	tokenDuration := time.Duration(config.RobotTokenDuration()) * time.Minute
 	expiresAt := time.Now().UTC().Add(tokenDuration).Unix()
+
 	createdName := common.RobotPrefix + robotReq.Name
 
+	if robotReq.ExpiresAt == 0 {
+		robotReq.ExpiresAt = expiresAt
```

Comment:
Suggest to move these two lines in this if statement.

```go	
tokenDuration := time.Duration(config.RobotTokenDuration()) * time.Minute
expiresAt := time.Now().UTC().Add(tokenDuration).Unix()
```

Reply:
ok